app/code/Magento/InventoryCatalog/Model/BulkPartialInventoryTransfer.php

```
@@ -0,0 +1,125 @@
+        $this->productIdsBySkus  = $getProductIdsBySkus;
+        $this->sourceItemsBySku  = $getSourceItemsBySkuAndSourceCodes;
+        $this->defaultSourceProvider = $defaultSourceProvider;
+        $this->sourceIndexer     = $sourceIndexer;
+        $this->legacyIndexer     = $legacyIndexer;
+    }
+
+    /**
+     * Run bulk partial inventory transfer for specified items.
+     *
+     * @param PartialInventoryTransferInterface[] $items
+     * @return SourceItemInterface[]
+     */
+    public function execute(array $items): array
+    {
+        $sources = [];
+        $processedSkus = [];
+        $sourceItems = [];
+
+        foreach ($items as $item) {
+            $validationResult = $this->transferValidator->validate($item);
+            if ($validationResult->isValid()) {
+                $this->transferCommand->execute($item);
+
+                $processedSkus[] = $item->getSku();
+                $sources[] = $item->getOriginSourceCode();
+                $sources[] = $item->getDestinationSourceCode();
+            }
+            $sourceItems += $this->sourceItemsBySku->execute($item->getSku(), [$item->getOriginSourceCode(), $item->getDestinationSourceCode()]);
+        }
+
+        $this->updateIndexes($sources, $processedSkus);
+        return $sourceItems;
+    }
+
+    /**
+     * @param string[] $sources
+     * @param string[] $skus
+     */
+    private function updateIndexes(array $sources, array $skus)
+    {
+        $sources = array_unique($sources);
+        $this->sourceIndexer->executeList($sources);
+
+        if (in_array($this->defaultSourceProvider->getCode(), $sources)) {
+            $this->updateLegacyIndex($skus);
+        }
+    }
+
+    /**
+     * @param string[] $skus
+     */
+    private function updateLegacyIndex(array $skus)
+    {
+        try {
+            $productIds = $this->productIdsBySkus->execute($skus);
+            $this->legacyIndexer->executeList($productIds);
+        } catch (NoSuchEntityException $e) {}
```

swallowing exceptions are usually bad practice , and if you do so, the description should be provided why